lib/broadside/deploy/ecs.rb

```diff
@@ -124,39 +128,58 @@ def bash
 
     private
 
-    def get_running_instance_ip_at_index(index)
-      ips = get_running_instance_ips
-      ips.size > index ? ips[index] : false
-    end
-
-    def get_running_instance_ip_at_index!(index)
-      ip = get_running_instance_ip_at_index(index)
-      return ip if ip
-      exception 'Instance index greater than available instance count.'
-    end
-
     # removes latest n task definitions
     def deregister_tasks(count)
-      get_task_def_ids.last(count).each do |td_id|
+      get_task_definition_arns.last(count).each do |td_id|
         ecs_client.deregister_task_definition({task_definition: td_id})
         debug "Deregistered #{td_id}"
       end
     end
 
     # creates a new task revision using current directory's env vars and provided tag
     def update_task
-      debug "Creating a new task definition..."
-      new_task_def = create_new_task_revision
+      unless get_latest_task_def_id
+        exception "No first task definition and cannot create one" unless @task_definition_config
 
-      container_def = new_task_def[:container_definitions].select { |c| c[:name] == family }.first
-      container_def[:environment] = @deploy_config.env_vars
-      container_def[:image] = image_tag
-      container_def[:command] = @deploy_config.command
```

Comment:
`@command` is an instance variable belonging to https://github.com/lumoslabs/broadside/blob/master/lib/broadside/configuration/deploy.rb#L45

`@deploy_config` belongs to this class, basically a copy of the configuration object above (https://github.com/lumoslabs/broadside/blob/master/lib/broadside/deploy.rb#L8). This was done so that an instance of a deploy would not corrupt data in the original configuration object.